projects/sandbox/datagen/datagen/scripts/waveforms.py

```diff
@@ -93,17 +95,32 @@ def main(
     prior, detector_frame_prior = prior()
     params = prior.sample(num_signals)
 
-    signals = generate_gw(
-        params,
-        minimum_frequency,
-        reference_frequency,
-        sample_rate,
-        waveform_duration,
-        waveform_approximant,
-        detector_frame_prior,
-    )
+    if signal_type == "bns":
+        with ProcessPoolExecutor(140) as exe:
+            future = exe.submit(
+                generate_gw_bns,
+                params,
+                minimum_frequency,
+                reference_frequency,
+                sample_rate,
+                waveform_duration,
+                waveform_approximant,
+                detector_frame_prior,
+            )
+            signals = future.result()
+    else:
```

Comment:
Since this is going into a dedicated `bns` branch, I think it might make sense to just assume we are generating bns waveforms, and not have this `if else`. We can move to generalizing everything down the line.

Reply:
If we don't want to create a separate repo for BNS, it will be most ideal and efficient, if we take care of this now rather than later. And write code keeping in mind that it works seamlessly with the main pipeline. Otherwise it will be a huge issue later to merge bns branch into main aframe branch.